File: Faturamento/Models/NotaFiscal.cs
namespace Faturamento.Models;

public class NotaFiscal
{
    public int Id { get; set; }

    public int Numero { get; set; }

    public StatusNotaFiscal Status { get; set; } = StatusNotaFiscal.ABERTA;

    public DateTime DataCriacao { get; set; }

    public List<ItemNotaFiscal> Itens { get; set; } = [];
}

File: frontend/src/app/models/nota-fiscal.ts
export type StatusNotaFiscal = 'ABERTA' | 'FECHADA';

export interface ItemNotaFiscal {
  id: number;
  notaFiscalId?: number;
  produtoId: number;
  quantidade: number;
}

export interface NotaFiscal {
  id: number;
  numero: number;
  status: StatusNotaFiscal;
  dataCriacao: string | Date;
  itens: ItemNotaFiscal[];
}

export interface ItemNotaFiscalRequest {
  produtoId: number;
  quantidade: number;
}

export interface CriarNotaFiscalRequest {
  itens: ItemNotaFiscalRequest[];
}
